third_party/iap_jwt_verify_nginx/src/jwt.cc

```diff
@@ -0,0 +1,209 @@
+// Copyright (C) 2002-2016 Igor Sysoev
```

Comment:
I looked into alternatives, but almost no C/C++ JWT libraries support ES256, the one I did find was neither actively maintained (last commit: 2 years ago) nor widely used, and there are trust issues with pulling in random code for a security feature. Not to mention you have to be careful about licensing. A JWT processing nginx module does exist, but it's 1) not open source or free (requires a commercial subscription), and 2) doesn't seem to have the functionality we want (no evidence that it supports ES256, doesn't have built-in audience validation so we'd still have to do that ourselves).

While rolling my own was the last thing I wanted to do, I felt it was the best choice after I investigated all the reuse options.